app/Services/FleetMissionService.php

```diff
@@ -68,42 +68,144 @@ public function __construct(PlayerService $player, MessageService $messageServic
+     *
+     * Returns:
+     * - int: The calculated fleet mission distance.
+     */
+    public function calculateFleetMissionDistance(PlanetService $fromPlanet, Coordinate $to): int
+    {
 
-        // Calculate distance between current planet and target planet.
-        // ----------------------------------------
-        // Between galaxies:
-        // 20.000 x (galaxy2 - galaxy1)
-        // Between systems:
-        // 2.700 + (95 x (system2 - system1))
-        // Between planets:
-        // 1.000 + (5 x (position2 - position1))
-        // Between moon or debris field and planet:
-        // 5
-        // ----------------------------------------
         $fromCoordinate = $fromPlanet->getPlanetCoordinates();
-        $distance = 0;
-        if ($fromCoordinate->galaxy !== $to->galaxy) {
-            $distance = 20000 * abs($to->galaxy - $fromCoordinate->galaxy);
+
+        $diffGalaxy = abs($fromCoordinate->galaxy - $to->galaxy);
+        $diffSystem = abs($fromCoordinate->system - $to->system);
+        $diffPlanet = abs($fromCoordinate->position - $to->position);
+
+        // Eğer galaksiler farklıysa
+        if ($diffGalaxy != 0) {
+            $diff2 = abs($diffGalaxy -  $this->settingsService->numberOfGalaxies());
+
+            if ($diff2 < $diffGalaxy) {
+                return $diff2 * 20000;
+            } else {
+                return $diffGalaxy * 20000;
+            }
+        }
+
+        // Eğer sistemler farklıysa
+        if ($diffSystem != 0) {
+            $diff2 = abs($diffSystem - 499);
+            $deltaSystem = 0;
+
+            if ($diff2 < $diffSystem) {
+                $deltaSystem = $diff2;
+            } else {
+                $deltaSystem = $diffSystem;
+            }
+
+            //deltaSystem = Math.max(deltaSystem - emptySystems - inactiveSystems, 1);
+            $deltaSystem = max($deltaSystem, 1);
+            return $deltaSystem * 5 * 19 + 2700;
         }
-        if ($fromCoordinate->system !== $to->system) {
-            $distance = 2700 + (95 * abs($to->system - $fromCoordinate->system));
+
+        // Eğer gezegenler farklıysa
+        if ($diffPlanet != 0) {
+            return $diffPlanet * 5 + 1000;
         }
-        if ($fromCoordinate->position !== $to->position) {
-            $distance = 1000 + (5 * abs($to->position - $fromCoordinate->position));
+
+        // Eğer koordinatlar aynıysa
+        return 5;
+    }
+
+    /**
+     * Calculate the consumption of a fleet mission based on the current planet, target coordinates and fleet.
+     * @param PlanetService $fromPlanet
+     * @param UnitCollection $ships
+     * @param Coordinate $target_coordinate
+     * @param int $holdingTime
+     * @param float $speed_percent
+     * @param $mission
+     * @return float|mixed
+     */
+    public function calcConsumption(PlanetService $fromPlanet, UnitCollection $ships, Coordinate $target_coordinate, int $holdingTime, float $speed_percent)
```

Comment:
```suggestion
    public function calculateConsumption(PlanetService $fromPlanet, UnitCollection $ships, Coordinate $target_coordinate, int $holdingTime, float $speed_percent)
```
This should probably be calculate, like the other methods :)